### PR TITLE
Fix stats underflow

### PR DESCRIPTION
**Pull request by nwf**
A proposed fix to #98

### CODE REVIEW

File: src/mem/allocstats.h [hunk 1]
```diff
@@ -183,6 +183,7 @@ namespace snmalloc
       UNUSED(sc);
 
 #ifdef USE_SNMALLOC_STATS
+      assert(sizeclass[sc].count.current > 0);
```
Comment (mjp41) [inline src/mem/allocstats.h]: Can't `current` become zero?  Should we make the statistics use a signed representation, and check for below 0?  Or perhaps, we could add an assert into `dec` that it is above 0?

Comment (nwf) [reply src/mem/allocstats.h]: This check is performed before the decrement, and so still permits `current` to become zero.  But yes, I agree that inside `dec()` is probably clearer; will move.

**Comment by mjp41** [comment]
Do you have a private test that caught this?  If so, it would be great to push it in, so I don't break stats in the future.

**Comment by nwf** [comment]
With the assertions in place, this was tripped by func-malloc.

**Pull request merged**